extensions/libarchive/tests/MergeFileTests.cpp

```diff
@@ -110,34 +112,34 @@ class FixedBuffer {
   }
 
  private:
-  std::unique_ptr<uint8_t[]> buf_;  // NOLINT(cppcoreguidelines-avoid-c-arrays)
+  std::vector<uint8_t> buf_;
   std::size_t size_ = 0;
   std::size_t capacity_ = 0;
 };
 
 std::vector<FixedBuffer> read_archives(const FixedBuffer& input) {
   class ArchiveEntryReader {
    public:
-    explicit ArchiveEntryReader(archive* arch) : arch(arch) {}
-    size_t read(std::span<std::byte> out_buffer) {
-      const auto ret = archive_read_data(arch, out_buffer.data(), out_buffer.size());
+    explicit ArchiveEntryReader(archive& arch) : arch(arch) {}
+    [[nodiscard]] size_t read(std::span<std::byte> out_buffer) const {
+      const auto ret = archive_read_data(&arch, out_buffer.data(), out_buffer.size());
       return ret < 0 ? minifi::io::STREAM_ERROR : gsl::narrow<size_t>(ret);
     }
    private:
-    archive* arch;
+    archive& arch;
   };
   std::vector<FixedBuffer> archive_contents;
-  struct archive *a = archive_read_new();
-  archive_read_support_format_all(a);
-  archive_read_support_filter_all(a);
-  archive_read_open_memory(a, input.begin(), input.size());
+  const auto a = minifi::processors::archive_read_unique_ptr{archive_read_new()};
```

Comment:
Rename this please as well

Reply:
👍 [review changes](https://github.com/apache/nifi-minifi-cpp/pull/1880/commits/3e748c09d8bd829d5e2b968ba9fda7108ae8f311)